Received Unprocessable Entity on version 1.0.0

I saw that a fix was applied for the Unprocessable Entity error but looks like I have received another variation. Running locally i receive no rubocop errors so I know I am not going over the limit of annotations.

```
/usr/bin/docker run --name d5970c2e4f0254d14a61063480604d82f_6a1e0b --label 10865d --workdir /github/workspace --rm -e GITHUB_TOKEN -e HOME -e GITHUB_REF -e GITHUB_SHA -e GITHUB_REPOSITORY -e GITHUB_ACTOR -e GITHUB_WORKFLOW -e GITHUB_HEAD_REF -e GITHUB_BASE_REF -e GITHUB_EVENT_NAME -e GITHUB_WORKSPACE -e GITHUB_ACTION -e GITHUB_EVENT_PATH -e RUNNER_OS -e RUNNER_TOOL_CACHE -e RUNNER_TEMP -e RUNNER_WORKSPACE -e ACTIONS_RUNTIME_URL -e ACTIONS_RUNTIME_TOKEN -e GITHUB_ACTIONS=true -v "/var/run/docker.sock":"/var/run/docker.sock" -v "/home/runner/work/_temp/_github_home":"/github/home" -v "/home/runner/work/_temp/_github_workflow":"/github/workflow" -v "/home/runner/work/ZuseAlpha/ZuseAlpha":"/github/workspace" 10865d:5970c2e4f0254d14a61063480604d82f
Building native extensions. This could take a while...
Successfully installed jaro_winkler-1.5.4
Successfully installed parallel-1.18.0
Successfully installed ast-2.4.0
Successfully installed parser-2.6.5.0
Successfully installed rainbow-3.0.0
Successfully installed ruby-progressbar-1.10.1
Successfully installed unicode-display_width-1.6.0
Successfully installed rubocop-0.76.0
Successfully installed rubocop-performance-1.5.0
Successfully installed rack-2.0.7
Successfully installed rubocop-rails-2.3.2
Successfully installed rubocop-minitest-0.3.0
Successfully installed rubocop-rspec-1.36.0
/action/lib/github_client.rb:36:in `request_http': Unprocessable Entity: {"message":"Invalid request.\n\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\nNo more than 50 items are allowed; 54 were supplied.","documentation_url":"https://developer.github.com/v3/checks/runs/#update-a-check-run"} (RuntimeError)
	from /action/lib/github_client.rb:10:in `patch'
	from /action/lib/github_check_run_service.rb:22:in `run'
	from /action/lib/index.rb:29:in `<main>'
##[error]Docker run failed with exit code 1
```

Please let me know if there is any more details needed or something I am missing.

--- a/lib/github_check_run_service.rb
+++ b/lib/github_check_run_service.rb
@@ -19,6 +19,7 @@ class GithubCheckRunService
     @annotations = @report_adapter.annotations(@report)
     @conclusion = @report_adapter.conslusion(@report)
 
+    puts update_check_payload
     @client.patch(
       "#{endpoint_url}/#{id}",
       update_check_payload

--- a/lib/github_check_run_service.rb
+++ b/lib/github_check_run_service.rb
@@ -47,14 +47,12 @@ class GithubCheckRunService
 
   def update_check_payload
     base_payload('completed').merge!(
-      {
-        conclusion: @conclusion,
-        output: {
-          title: CHECK_NAME,
-          summary: @summary,
-          annotations: (@annotations if @conclusion == 'failure')
-        }.compact
-      }
+      conclusion: @conclusion,
+      output: {
+        title: CHECK_NAME,
+        summary: @summary,
+        annotations: (@annotations if @conclusion == 'failure')
+      }.compact
     )
   end
 end

--- a/lib/github_check_run_service.rb
+++ b/lib/github_check_run_service.rb
@@ -19,7 +19,6 @@ class GithubCheckRunService
     @annotations = @report_adapter.annotations(@report)
     @conclusion = @report_adapter.conslusion(@report)
 
-    puts update_check_payload
     @client.patch(
       "#{endpoint_url}/#{id}",
       update_check_payload

--- a/lib/github_check_run_service.rb
+++ b/lib/github_check_run_service.rb
@@ -52,8 +52,9 @@ class GithubCheckRunService
         output: {
           title: CHECK_NAME,
           summary: @summary,
-        }.merge!({ annotations: @annotations } if @conclusion == 'failure')
-      }
+          annotations: (@annotations if @conclusion == 'failure')
+        }
+      }.compact
     )
   end
 end

--- a/lib/github_check_run_service.rb
+++ b/lib/github_check_run_service.rb
@@ -32,27 +32,28 @@ class GithubCheckRunService
     "/repos/#{@github_data[:owner]}/#{@github_data[:repo]}/check-runs"
   end
 
-  def create_check_payload
+  def base_payload(status)
     {
       name: CHECK_NAME,
       head_sha: @github_data[:sha],
-      status: 'in_progress',
+      status: status,
       started_at: Time.now.iso8601
     }
   end
 
+  def create_check_payload
+    base_payload('in_progress')
+  end
+
   def update_check_payload
-    {
-      name: CHECK_NAME,
-      head_sha: @github_data[:sha],
-      status: 'completed',
-      completed_at: Time.now.iso8601,
-      conclusion: @conclusion,
-      output: {
-        title: CHECK_NAME,
-        summary: @summary,
-        annotations: @annotations
+    base_payload('completed').merge!(
+      {
+        conclusion: @conclusion,
+        output: {
+          title: CHECK_NAME,
+          summary: @summary,
+        }.merge!({ annotations: @annotations } if @conclusion == 'failure')
       }
-    }
+    )
   end
 end

--- a/lib/github_check_run_service.rb
+++ b/lib/github_check_run_service.rb
@@ -53,8 +53,8 @@ class GithubCheckRunService
           title: CHECK_NAME,
           summary: @summary,
           annotations: (@annotations if @conclusion == 'failure')
-        }
-      }.compact
+        }.compact
+      }
     )
   end
 end

